velox/functions/sparksql/SIMDJsonFunctions.h

```diff
@@ -0,0 +1,192 @@
+    if (jsonPath != nullptr) {
+      formattedJsonPath_ = getFormattedJsonPath(*jsonPath);
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE simdjson::error_code extractStringResult(
+      simdjson_result<ondemand::value> rawResult,
+      out_type<Varchar>& result) {
+    simdjson::error_code error;
+    std::stringstream ss;
+    switch (rawResult.type()) {
+      // For number and bool types, we need to explicitly get the value
+      // for specific types instead of using `ss << rawResult`. Thus, we
+      // can make simdjson's internal parsing position moved and then we
+      // can check the validity of ending character.
+      case ondemand::json_type::number: {
+        switch (rawResult.get_number_type()) {
+          case ondemand::number_type::unsigned_integer: {
+            uint64_t numberResult;
+            error = rawResult.get_uint64().get(numberResult);
+            if (!error) {
+              ss << numberResult;
+              result.append(ss.str());
+            }
+            return error;
+          }
+          case ondemand::number_type::signed_integer: {
+            int64_t numberResult;
+            error = rawResult.get_int64().get(numberResult);
+            if (!error) {
+              ss << numberResult;
+              result.append(ss.str());
+            }
+            return error;
+          }
+          case ondemand::number_type::floating_point_number: {
+            double numberResult;
+            error = rawResult.get_double().get(numberResult);
+            if (!error) {
+              ss << numberResult;
+              result.append(ss.str());
+            }
+            return error;
+          }
+          default:
+            VELOX_UNREACHABLE();
+        }
+      }
+      case ondemand::json_type::boolean: {
+        bool boolResult = false;
+        rawResult.get_bool().get(boolResult);
+        if (boolResult) {
+          result.append("true");
+        } else {
+          result.append("false");
+        }
+        return SUCCESS;
+      }
+      case ondemand::json_type::string: {
+        std::string_view stringResult;
+        error = rawResult.get_string().get(stringResult);
+        result.append(stringResult);
+        return error;
+      }
+      case ondemand::json_type::object: {
+        // For nested case, e.g., for "{"my": {"hello": 10}}", "$.my" will
+        // return an object type.
+        ss << rawResult;
+        result.append(ss.str());
+        return SUCCESS;
+      }
+      case ondemand::json_type::array: {
+        ss << rawResult;
+        result.append(ss.str());
+        return SUCCESS;
+      }
+      default: {
+        return UNSUPPORTED_ARCHITECTURE;
+      }
+    }
+  }
+
+  // This is a simple validation by checking whether the obtained result is
+  // followed by valid char. Because ondemand parsing we are using ignores json
+  // format validation for characters following the current parsing position.
+  bool isValidEndingCharacter(const char* currentPos) {
```

Comment:
Are we ensured all valid characters are covered?

Reply:
Is this Spark-specific? Do we not need same logic for Presto?

CC: @Yuhta